src/System.Memory/src/System/SpanSortHelpers.Keys.Comparison.cs

```diff
@@ -0,0 +1,329 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+
+#if !netstandard
+using Internal.Runtime.CompilerServices;
+#endif
+
+using static System.SpanSortHelpersCommon;
+
+namespace System
+{
+    internal static partial class SpanSortHelpersKeys
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static void Sort<TKey>(
+            ref TKey keys, int length,
+            Comparison<TKey> comparison)
+        {
+            IntrospectiveSort(ref keys, length, comparison);
+        }
+
+        private static void IntrospectiveSort<TKey>(
+            ref TKey keys, int length,
+            Comparison<TKey> comparison)
+        {
+            var depthLimit = 2 * FloorLog2PlusOne(length);
+            IntroSort(ref keys, 0, length - 1, depthLimit, comparison);
+        }
+
+        private static void IntroSort<TKey>(
+            ref TKey keys, 
+            int lo, int hi, int depthLimit,
+            Comparison<TKey> comparison)
+        {
+            Debug.Assert(comparison != null);
+            Debug.Assert(lo >= 0);
+
+            while (hi > lo)
+            {
+                int partitionSize = hi - lo + 1;
+                if (partitionSize <= IntrosortSizeThreshold)
+                {
+                    if (partitionSize == 1)
+                    {
+                        return;
+                    }
+                    if (partitionSize == 2)
+                    {
+                        Sort2(ref keys, lo, hi, comparison);
+                        return;
+                    }
+                    if (partitionSize == 3)
+                    {
+                        ref TKey loRef = ref Unsafe.Add(ref keys, lo);
+                        ref TKey miRef = ref Unsafe.Add(ref keys, hi - 1);
+                        ref TKey hiRef = ref Unsafe.Add(ref keys, hi);
+                        //ref TKey miRef = ref Unsafe.SubtractByteOffset(ref hiRef, new IntPtr(Unsafe.SizeOf<TKey>()));
```

Comment:
Remove commented out code if not necessary. I see in other places it acts as documentation. Those are fine.